packages/web-components/fast-components/src/menu/menu.ts

```diff
@@ -0,0 +1,181 @@
+import { DOM, FASTElement, observable } from "@microsoft/fast-element";
+import { inRange, invert } from "lodash-es";
+import {
+    isHTMLElement,
+    keyCodeArrowDown,
+    keyCodeArrowLeft,
+    keyCodeArrowRight,
+    keyCodeArrowUp,
+    keyCodeEnd,
+    keyCodeHome,
+} from "@microsoft/fast-web-utilities";
+import { MenuItemRole } from "../menu-item";
+
+export class Menu extends FASTElement {
+    @observable
+    public items: HTMLSlotElement;
+    private itemsChanged(): void {
+        this.menuItems = this.domChildren();
+    }
+
+    private menuItems: Element[];
+
+    /**
+     * The index of the focusable element in the items array
+     * defaults to -1
+     */
+    private focusIndex: number = -1;
+
+    private static focusableElementRoles: { [key: string]: string } = invert(
+        MenuItemRole
+    );
+
+    public connectedCallback(): void {
+        super.connectedCallback();
+
+        // We need to queue update currently to ensure that
+        // the attributes on our menu items have time to be set
```

Comment:
This seems like race-case territory - is it possible for these elements to *not* have the data needed when this `queueUpdate` callback gets called?

Reply:
Yes - I spoke with @EisenbergEffect previously and I think we wanted to find a better way to handle this.

Reply:
Is that something we want to take for this iteration? The other case this doesn't address is items added after connection.

Reply:
It seems relatively low cost - let’s take it here. FYI @eljefe223

Reply:
We should be good now.